src/student/student.controller.ts

```diff
@@ -1,19 +1,54 @@
-import { Body, Controller, Get, Inject, Param, ParseUUIDPipe, Put, UseGuards, } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  DefaultValuePipe,
+  Get,
+  Inject,
+  Param,
+  ParseEnumPipe,
+  ParseIntPipe,
+  ParseUUIDPipe,
+  Put,
+  Query,
+  UseGuards,
+} from '@nestjs/common';
 import { StudentService } from './student.service';
-import { GetSingleStudentFullDetailsResponse, OneStudentResponse, StudentStatus, } from '../types';
+import {
+  GetSingleStudentFullDetailsResponse,
+  ItemsPerSite,
+  OneStudentResponse,
+  StudentsListResponse,
+  StudentStatus
+} from '../types';
 import { UpdateStudentDetailsDto } from './dto/update-student-details.dto';
 import { UpdateStudentDetailsResponse } from '../types/student/update-student-details-response';
 import { CheckUniquePropertiesGuard } from '../guards/check-unique-properties.guard';
 import { ChangeStudentStatusResponse } from '../types/student/change-student-status-response';
-import { CheckScheduledStudentsLimitGuard } from "../guards/check-scheduled-students-limit.guard";
+import { CheckScheduledStudentsLimitGuard } from '../guards/check-scheduled-students-limit.guard';
 
 @Controller('student')
 export class StudentController {
   constructor(@Inject(StudentService) private studentService: StudentService) {}
 
   @Get('/available-students')
-  async getAvailableStudents(): Promise<Omit<OneStudentResponse, 'degrees'>[]> {
-    return await this.studentService.getStudents(StudentStatus.available);
+  async getAvailableStudents(
+      @Query(
+          'itemsPerSite',
+          new DefaultValuePipe(10),
+          ParseIntPipe,
+          new ParseEnumPipe(ItemsPerSite),
+      )
+          itemsPerSite: ItemsPerSite,
+      @Query('pageNo', new DefaultValuePipe(1), ParseIntPipe) pageNO: number,
+      @Query('city', new DefaultValuePipe('')) city: string,
+  ): Promise<StudentsListResponse> {
+    console.log(itemsPerSite);
```

Comment:
You can remove this console log

Reply:
@F-Wojciechowski it's already removed. Do you approve?